libcst/metadata/scope_provider.py

```diff
@@ -284,6 +292,14 @@ def f(self) -> "c":
                 )
         return results
 
+    def get_assignments_for(self, node: cst.CSTNode) -> Collection[BaseAssignment]:
+        """ Get all assignments in current scope given a :class:`~libcst.CSTNode`. """
+        full_name = _QualifiedNameUtil.get_full_name_for(node)
+        if full_name is None:
+            return set()
+        parts = full_name.split(".")
+        return self[parts[0]]
```

Comment:
I'm not sure on the implications of claiming that "a.b" assignments are equivalent to "a" assignments. Do we have this documented anywhere?

Reply:
We decided to use Assignments class to replace this function and we'll clarify the current scope analysis only handles local variable name.